packages/react-reconciler/src/ReactFiberCommitWork.new.js

```diff
@@ -1742,8 +1924,19 @@ function commitWork(current: Fiber | null, finishedWork: Fiber): void {
     case OffscreenComponent:
     case LegacyHiddenComponent: {
       const newState: OffscreenState | null = finishedWork.memoizedState;
+      const newProps = finishedWork.memoizedProps;
```

Comment:
These changes should only be made `OffscreenComponent`. Not `LegacyHiddenComponent`.

Also the whole thing can be wrapped in the feature flag, so it gets stripped out of the non-FB builds.